### PR TITLE
Feature/ba-0010

### PR DESCRIPTION
**Pull request by IMozgg**
added feature:
- new entity POI;
- new interface DAO for POI;
- new interface Service and Controller for POI.
fix:
- format LocalDate for column startWork;
- deny input negative value for column workTime.

### CODE REVIEW

**Comment by IMozgg** [comment]
resolve all conflicts and change entity Poi to PointOfInterest

File: rest/src/main/java/io/khasang/ba/controller/PointOfInterestController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import io.khasang.ba.service.PointOfInterestService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/poi")
+public class PoiController {
```
Comment (SKuznet) [inline rest/src/main/java/io/khasang/ba/controller/PointOfInterestController.java]: change name to full

File: rest/src/main/java/io/khasang/ba/entity/PointOfInterest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+/**
+ *
+ * POI - Point of Interest. This is object include any params i.e. address, geographical coordinates,
+ * include category etc.
+ */
+
+@Entity
+@Table(name = "poi")
+public class Poi {
```
Comment (SKuznet) [inline rest/src/main/java/io/khasang/ba/entity/PointOfInterest.java]: change to full

File: rest/src/main/java/io/khasang/ba/entity/PointOfInterest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    private String name;
+
+    @ColumnDefault(value = "'безымянный'")
```
Comment (SKuznet) [inline rest/src/main/java/io/khasang/ba/entity/PointOfInterest.java]: change ru to en

File: rest/src/main/java/io/khasang/ba/service/PointOfInterestService.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package io.khasang.ba.service;
+
+import io.khasang.ba.entity.Poi;
+
+import java.util.List;
+
+public interface PoiService {
```
Comment (SKuznet) [inline rest/src/main/java/io/khasang/ba/service/PointOfInterestService.java]: change to full  name

File: rest/src/main/java/io/khasang/ba/service/impl/PointOfInterestServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+@Service
+public class PoiServiceImpl implements PoiService {
+
+    @Autowired
+    private PoiDao poiDao;
+
+    @Override
+    public Poi addPoi(Poi poi) {
+        poiDao.add(poi);
```
Comment (SKuznet) [inline rest/src/main/java/io/khasang/ba/service/impl/PointOfInterestServiceImpl.java]: simplify

File: rest/src/main/java/io/khasang/ba/service/impl/PointOfInterestServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    public Poi updatePoi(Poi poi) {
+        return poiDao.update(poi);
+    }
+
+    @Override
+    public Poi deletePoi(long id) {
+        Poi temp = poiDao.getById(id);
```
Comment (SKuznet) [inline rest/src/main/java/io/khasang/ba/service/impl/PointOfInterestServiceImpl.java]: simplify

**Review by SKuznet — CHANGES_REQUESTED** [review]

**Pull request merged**